docs/pages/reference/access-controls/roles.mdx

```diff
@@ -405,7 +404,165 @@ spec:
       where: contains(session.participants, user.metadata.name)
 ```
 
-Role for restricted access to active sessions:
+### Filter fields
+
+The `where` field in a rule allows you to filter access to resources based on
+user traits and resource metadata.
+
+#### User fields
+
+The following user fields can be used in `where` expressions to filter access.
+The `user` object represents the authenticated user making the request to
+access the session recording.
+
+| User Field | Description | Type |
+|---|---|---|
+| `user.metadata.name` | The name of the user. | `string` |
+| `user.spec.traits` | The traits assigned to the user. | `dict<string, array<string>>` |
+| `user.spec.roles` | The roles assigned to the user. | `array<string> `|
+
+Examples:
+
+```yaml
+where: contains(user.spec.traits["team"], "devops")
+where: contains(user.spec.roles, "admin")
+```
+
+### Session fields
+
+<Admonition
+  type="warning"
+  title="Session fields are immutable">
+
+Session fields are immutable and capture the state of the session at the moment it
+was created.
+
+If the underlying resource changes (e.g., server labels), these fields will not be
+updated, and the session’s RBAC will not reflect the changes.
+
+</Admonition>
+
+The following session fields can be used in `where` expressions to filter access.
+The `session` object represents the session resource being accessed.
+It contains metadata about the session, including the participants, protocol,
+resource type and the session creator (user who initiated the session) roles and traits.
+
+| Session Field | Description | Type |
+|---|---|---|
+| `session.id` | The unique identifier of the session. | `string` |
+| `session.kind` | The kind of session (e.g., "ssh", "kube", "db", "rdp"). | `string` |
+| `session.participants` | The list of users who participated in the session. | `array<string>` |
+| `session.protocol` | The protocol used for the session (e.g., "ssh", "kube", "db", "rdp"). | `string` |
+| `session.user` | The user who initiated the session. | `string` |
+| `session.user_roles` | The roles assigned to the user who initiated the session. | `array<string>` |
+| `session.user_traits` | The traits assigned to the user who initiated the session. | `dict<string, array<string>>` |
+
+Examples:
+
+```yaml
+# List only sessions initiated by the user
+where: session.user == user.metadata.name
+# List sessions where the user has the "devops" team trait
+where: contains(session.user_traits["team"], "devops")
+# List sessions initiated by users with the "admin" role
+where: contains(session.user_roles, "admin")
```

Comment:
I attempted to test with a role in 18.1.8 for limiting session recordings. Are these docs for that version or a upcoming version?

```yaml
    - resources:
      - session
      verbs:
      - read
      - list
      where: contains(session.user_roles, "access")
```

I get this error on the session recordings page for the users. 

<img width="624" height="296" alt="image" src="https://github.com/user-attachments/assets/5a0e8643-5fa4-4e74-9c08-d75f0b3b027b" />